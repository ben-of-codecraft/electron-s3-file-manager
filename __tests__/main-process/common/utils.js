const utils = require('../../../src/main-process/common/utils');

describe('encrypt', () => {
	test('encrypt string with iv', () => {
		const result = utils.encrypt({
			value: Buffer.from('test value'),
			iv: Buffer.from('31682eec1f9655da345101e94ca741ba', 'hex'),
		});

		expect(result.toString('hex')).toMatchSnapshot();
	});

	test('encrypt string with invalid iv', () => {
		expect(() => {
			utils.encrypt({
				value: Buffer.from('test value'),
				iv: Buffer.from('31682eec1f9655da345101e94ca741', 'hex'),
			});
		}).toThrow('Invalid initialization vector');
	});
});

describe('decrypt', () => {
	test('decrypt string with iv', () => {
		const result = utils.decrypt({
			value: Buffer.from('8595b720aebb469f8ce1cb1c5982a813', 'hex'),
			iv: Buffer.from('31682eec1f9655da345101e94ca741ba', 'hex'),
		});

		expect(result.toString()).toMatchSnapshot();
	});

	test('encrypt string with invalid iv', () => {
		expect(() => {
			utils.decrypt({
				value: Buffer.from('8595b720aebb469f8ce1cb1c5982a813', 'hex'),
				iv: Buffer.from('31682eec1f9655da345101e94ca741', 'hex'),
			});
		}).toThrow('Invalid initialization vector');
	});
});

describe('parse keyword', () => {
	test('parse keyword', () => {
		const keywords = utils.parseKeyword('car "apple pencil" -amd uId:1024 aId: 2048');

		expect(keywords).toMatchSnapshot();
	});
});
