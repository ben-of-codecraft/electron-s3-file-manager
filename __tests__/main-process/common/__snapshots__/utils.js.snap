// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decrypt decrypt string with iv 1`] = `"test value"`;

exports[`encrypt encrypt string with iv 1`] = `"8595b720aebb469f8ce1cb1c5982a813"`;

exports[`parse keyword parse keyword 1`] = `
{
  "fields": {
    "aId": "2048",
    "uId": "1024",
  },
  "minus": [
    "amd",
  ],
  "plus": [
    "apple pencil",
    "car",
  ],
}
`;
